controllers/users.js

```diff
@@ -276,18 +277,78 @@ const updateSelf = async (req, res) => {
  * @param res {Object} - Express response object
  */
 const postUserPicture = async (req, res) => {
+  const { file } = req;
+  const { id: userId, discordId } = req.userData;
+  const { coordinates } = req.body;
+  let discordAvatarUrl = "";
+  let imageData;
+  let verificationResult;
+  try {
+    discordAvatarUrl = await generateDiscordProfileImageUrl(discordId);
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
   try {
-    const { file } = req;
-    const { id: userId } = req.userData;
-    const { coordinates } = req.body;
     const coordinatesObject = coordinates && JSON.parse(coordinates);
-    const imageData = await imageService.uploadProfilePicture({ file, userId, coordinates: coordinatesObject });
+    imageData = await imageService.uploadProfilePicture({ file, userId, coordinates: coordinatesObject });
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+  try {
+    verificationResult = await userQuery.addForVerification(userId, discordId, imageData.url, discordAvatarUrl);
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+  return res.status(201).json({
+    message: "Profile picture uploaded successfully! " + verificationResult.message,
```

Comment:
Nit : we can use ES6 Template literal for string concatenation.

Reply:
Cool. Will change it